### PR TITLE
Add JCB capability request notice

### LINKED ISSUE #6804: Add a notice for JCB capability request on the Settings page
### Description
JCB is a card payment method that needs a separate capability. To activate JCB in their stores, Japanese merchants must provide additional information to get the capability approved. To be able to request the capability, a notice above the payment methods list needs to be provided to the merchants.

enable-jcb-1

### Acceptance criteria
- A notice must be displayed when JCB capability is `unrequested` and the account country is Japan.
- The notice must guide the merchants through all capability statuses: `unrequested`, `pending`, `active`, `inactive`
- The notice must be dismissable with a confirmation popup.

### Designs
https://www.figma.com/file/C1r0Vzxh1mhbDJJnbjHp5q/Payments-Settings?type=design&node-id=2622%3A33711&mode=design&t=L4MM6rrMQwyS8Txl-1

### PR DESCRIPTION
**Pull request by mordeth**
Fixes https://github.com/Automattic/woocommerce-payments/issues/6804

#### Changes proposed in this Pull Request
Add a notice to request JCB capability above Payment Methods.

Screenshot 2023-09-22 at 0 43 53

#### Testing instructions
*Case 1*
- Connect a Japanese account.
- Navigate to `Payments` -> `Settings`.
- Scroll down to Payment Methods.
- Verify the notice is displayed above Payment Methods.
- Click the `Enable JCB` button.
- Verify the notice disappears.
- Refresh the page and navigate to Payment Methods.
- Verify the capability is requested and the notice text is changed.
Screenshot 2023-09-22 at 0 48 10

*Case 2*
- Connect a Japanese account.
- Navigate to `Payments` -> `Settings`.
- Scroll down to Payment Methods.
- Verify the notice is displayed above Payment Methods.
- Click the dismiss icon on the top right corner.
- Verify a modal is displayed warning you the notice won't be shown again
- Click `Yes, continue`
- Verify the notice is hidden.
- Refresh the page
- Verify the notice is not displayed

Screenshot 2023-09-22 at 0 49 51

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7224` or branch name `add/jcb-capability-request` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/jcb-capability-request) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 5ef92752a8f067cd0c7b434938a7b668847e7487
- Build time: 2023-10-04 06:01:00 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-methods/capability-request/capability-request-dismiss-modal.tsx
Comment (dpaun1985) [inline client/payment-methods/capability-request/capability-request-dismiss-modal.tsx]: a typo here `confirme`

File: client/payment-methods/capability-request/capability-request-map.ts
Comment (dpaun1985) [inline client/payment-methods/capability-request/capability-request-map.ts]: I think the `actionUrl` is wrong.

File: includes/admin/class-wc-rest-payments-settings-controller.php
Comment (dpaun1985) [inline includes/admin/class-wc-rest-payments-settings-controller.php]: we can merge these ifs

File: includes/admin/class-wc-rest-payments-settings-controller.php
Comment (dpaun1985) [inline includes/admin/class-wc-rest-payments-settings-controller.php]: we can add this block under `if ( 'unrequested' === $payment_method_statuses[ $stripe_key ]['status'] ) {`, and remove `$cache_needs_refresh     = false;` line

**Review by dpaun1985 — COMMENTED** [review]
Looks good overall. Left a few comments.

File: client/payment-methods/capability-request/capability-request-dismiss-modal.tsx
Comment (anu-rock) [reply client/payment-methods/capability-request/capability-request-dismiss-modal.tsx]: There's another typo `chooising`

File: changelog/add-jcb-capability-request
Comment (anu-rock) [inline changelog/add-jcb-capability-request]: It sounds a bit too jargony. What do you think about:

```suggestion
Add a notice on the Settings page to request JCB capability for Japanese customers.
```

File: client/payment-methods/capability-request/capability-request-dismiss-modal.tsx
Comment (anu-rock) [inline client/payment-methods/capability-request/capability-request-dismiss-modal.tsx]: Calling JCB a "payment method" might be confusing for the merchants.

File: client/payment-methods/capability-request/capability-request-dismiss-modal.tsx
Comment (anu-rock) [reply client/payment-methods/capability-request/capability-request-dismiss-modal.tsx]: Here's my attempt at a refined (and hopefully grammatically correct) message:

> Choosing to continue will remove the option to accept JCB cards from your customers. The option to enable JCB will not appear again.

File: client/payment-methods/capability-request/capability-request-map.ts
Comment (anu-rock) [inline client/payment-methods/capability-request/capability-request-map.ts]: Calling JCB a "payment method" might be confusing for the merchants.

File: client/payment-methods/capability-request/capability-request-notice.tsx
Comment (anu-rock) [inline client/payment-methods/capability-request/capability-request-notice.tsx:19]: There's a JS linting error here.

Screenshot 2023-09-26 at 4 49 06 PM

**Review by anu-rock — COMMENTED** [review]
I wasn't able to do a deep-dive code review. But I've nonetheless left a few comments.

Manual testing of Case 1 worked as expected:

#### Step 1

Screenshot 2023-09-26 at 4 36 46 PM

#### Step 2

Screenshot 2023-09-26 at 4 37 26 PM

___

⚠️ Clicking the "Enable JCB" button did not provide any feedback. For 3-4 secs, nothing seemed to happen until the success notification appeared in bottom left.

⚠️ Since this PR requires two fresh Japanese accounts to test your two cases, it will be helpful to mention sample profile data for the onboarding process. I ended up spending a lot of time fishing for random Japanese names in kanji and kana; the same for random addresses.

File: client/payment-methods/capability-request/capability-request-dismiss-modal.tsx
Comment (mordeth) [reply client/payment-methods/capability-request/capability-request-dismiss-modal.tsx]: Thank you, the message is updated!

File: client/payment-methods/capability-request/capability-request-map.ts
Comment (mordeth) [reply client/payment-methods/capability-request/capability-request-map.ts]: Adam provided the correct URL and updated it.

File: includes/admin/class-wc-rest-payments-settings-controller.php
Comment (mordeth) [reply includes/admin/class-wc-rest-payments-settings-controller.php]: Good suggestion, updated!

File: client/payment-methods/capability-request/capability-request-dismiss-modal.tsx
Comment (mordeth) [reply client/payment-methods/capability-request/capability-request-dismiss-modal.tsx]: Makes sense, updated it!

File: client/payment-methods/capability-request/capability-request-map.ts
Comment (mordeth) [reply client/payment-methods/capability-request/capability-request-map.ts]: Updated!

File: client/payment-methods/capability-request/capability-request-notice.tsx
Comment (mordeth) [reply client/payment-methods/capability-request/capability-request-notice.tsx:19]: Thank you! Resolved.

**Comment by mordeth** [comment]
>Clicking the "Enable JCB" button did not provide any feedback. For 3-4 secs, nothing seemed to happen until the success notification appeared in bottom left.

@anu-rock Good catch, I decided to extend the `InlineNotice` component to accept `isBusy` param and now when the Request button is clicked a loading state is displayed and the button is disabled.

File: changelog/add-jcb-capability-request
Comment (mordeth) [reply changelog/add-jcb-capability-request]: Thank you, updated!

**Review by dpaun1985 — APPROVED** [review]
✅ LGTM - code review only

**Comment by anu-rock** [comment]
> @anu-rock Good catch, I decided to extend the InlineNotice component to accept isBusy param and now when the Request button is clicked a loading state is displayed and the button is disabled.

@mordeth Thanks for the improvement :)

File: client/payment-methods/capability-request/capability-request-map.ts
Comment (anu-rock) [inline client/payment-methods/capability-request/capability-request-map.ts]:
```suggestion
					'JCB is now enabled on your store.',
```

**Review by anu-rock — APPROVED** [review]
There's one open comment from Dan that should be resolved before merging, otherwise LGTM after the latest changes 👍 

P.S. I didn't do another round of manual testing.

**Pull request merged**
